packages/eslint-plugin/docs/rules/no-extraneous-class.md

```diff
@@ -2,14 +2,21 @@
 
 Forbids the use of classes as namespaces.
 
-This rule warns when a class is accidentally used as a namespace.
+This rule warns when a class has no non-static members, such as for a class used exclusively as a static namespace.
 
 ## Rule Details
 
-From TSLint’s docs:
+Users who come from a [OOP](https://en.wikipedia.org/wiki/Object-oriented_programming) paradigm may wrap their utility functions in an extra class, instead of putting them at the top level of an ECMAScript module.
+Doing so is generally unnecessary in JavaScript and TypeScript projects.
 
-> Users who come from a Java-style OO language may wrap their utility functions in an extra class,
-> instead of putting them at the top level.
+- Wrapper classes add extra runtime bloat and cognitive complexity to code without adding any structural improvements
+  - Whatever would be put on them, such as utility functions, are already organized by virtue of being in a module.
+  - As an alternative, you can always `import * as ...` the module to get all of them in a single object.
+- IDEs can't provide as good autocompletions when you start typing the names of the helpers, since they're on a class instead of freely available to import
```

Comment:
@goktugyil https://github.com/typescript-eslint/typescript-eslint/issues/5040#issuecomment-1134013764:

> What situation does IDEs not provide good autocompletions?

Take [this TypeScript playground repro with a `Settings` class and static `version` field](https://www.typescriptlang.org/play?#code/MYGwhgzhAEDKCmAXRBLAdgcxgbwFDQOgkTFWGgDd4AnCFAezWgF5oByARgDoAmLgZjYBuXAF9cuYIwj0Q8LiHoYAFAAMAsgE9KNOo2goYAEmxVqooA). Typing out `ver` won't provide autocompletes even though the `Settings` class is in the same file.

![TypeScript playground with a `Settings` class and static `version` field, where typing 'ver' does not autocomplete to `Settings.version`](https://user-images.githubusercontent.com/3335181/169891883-866d9a92-f429-4b0b-9c6d-7d19b402bd44.png)

On the other hand, if `version` were declared as a variable -and even if it's exported from another file-, TypeScript can autocomplete to it. Take [this TypeScript playground repro with a `version` const](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAbgUwE4QJbhgXhgcgIwB0ATAQMw4DcAUFaJCADYIEMgDmAFAAYCyAnvGRoMqCDAAkAb0RIAvkA).

![TypeScript playground with a `version` const, where typing 'ver' autocompletes to `version`](https://user-images.githubusercontent.com/3335181/169892170-6685a967-4ceb-4d6e-a678-31db1c42a45b.png)

Reply:
I said in another comment too but, I think in this context, it would be more appropriate to compare `namepace.version` to `staticClass.version`. As the reason of using static class is organizing the variable under a namespace, instead of leaving it hanging around.

And nothing changes in terms of autocomplete when you turn it into a namespace. Also you'll have to add another word, `export`